gordo_components/model/models.py

```diff
@@ -549,26 +437,31 @@ def fit(
                 f"target, but a y was supplied. It will not be used."
             )
 
-        self._validate_size_of_X(X)
-        gen = self.generate_window(X, output_y=True)
-
-        if not hasattr(self, "model"):
-            # these are only used for the intermediate fit method (fit method of KerasBaseEstimator),
-            # called only to initiate the model
-            super().fit(*next(gen), epochs=1, verbose=0, **kwargs)
-
-        steps_per_epoch = self.calc_steps_per_epoch(X)
-
-        self.kwargs["steps_per_epoch"] = steps_per_epoch
+        X = self._validate_and_fix_size_of_X(X)
 
+        # We call super.fit on a single sample (notice the batch_size=1) to initiate the
+        # model using the scikit-learn wrapper.
+        tsd = get_TimeseriesGenerator(
+            data=X, batch_size=1, lookback_window=self.lookback_window, lookahead=0
```

Comment:
Would it be more efficient to just send in a slice of X? Since we only need a single sample? I guess not much point, since it doesnt actually do much before the batch is yielded...

Reply:
it would. I did that. Good tip